orchestra/proc-macro/src/impl_orchestra.rs

```diff
@@ -91,9 +91,9 @@ pub(crate) fn impl_orchestra_struct(info: &OrchestraInfo) -> proc_macro2::TokenS
 			>,
 
 			/// Gather running subsystems' outbound streams into one.
-			to_orchestra_rx: #support_crate ::stream::Fuse<
+			to_orchestra_rx: Option<#support_crate ::stream::Fuse<
```

Comment:
why did this become an `Option`?

Reply:
So the reason is you need to move the receiver to another future task, and therefore - since `stop` and all the other associated functions require `self` (non referential) as arguments, you have to be able to retain `Self` as type (no unpacking) due to the above as well.

I am not a particular fan of using `Option<_>`, it obfuscates the rationale. Either we add a new type, or we create an `<OrchestraName>Inner` type that is only the part that is being spawned and one then can unpack the `<OrchestraName>` into the inner type plus receivers.

Not sure there is a better alternative from the top of my head.